interfaces/apparmor/template_vars.go

```diff
@@ -88,22 +97,23 @@ func modernVariables(appInfo *snap.AppInfo) string {
 // profile. That same content also decides if the profile is enforcing or
 // advisory (complain). This is used to implement developer mode.
 func (b *Backend) aaHeader(appInfo *snap.AppInfo, developerMode bool) []byte {
-	var text string
-	// Use a different template when requested.
-	if len(b.CustomTemplate) == 0 {
-		text = strings.TrimRight(defaultTemplate, "\n}")
-		// TODO: switch to modernVariables() after the default template is
-		// compatible with them.
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
-	} else {
-		text = strings.TrimRight(b.CustomTemplate, "\n}")
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
+	template := b.legacyTemplate
+	if template == nil {
+		template = defaultTemplate
 	}
+	template = bytes.TrimRight(template, "\n}")
```

Comment:
Per the earlier review note, let's please not trim the curly brace off. I believe there's no reason to fiddle with the template provided. Let's just inject the content where due.